pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,303 @@
+                    Ok(Some(PendingTx {
+                        did: to_did,
+                        bridge_tx,
+                    })) => {
+                        let entry = new_pending_txs
+                            .entry(to_did)
+                            .or_default();
+                        entry.push(bridge_tx.clone());
+                        Self::deposit_event(RawEvent::Pending(PendingTx {
+                            did: to_did,
+                            bridge_tx
+                        }));
+                    }
+                    Err(_) => Self::deposit_event(RawEvent::Failed(bridge_tx)),
+                }
+            }
+            for (to_did, txs) in new_pending_txs {
+                if to_did == did {
+                    <PendingTxs<T>>::insert(did, txs);
+                } else {
+                    <PendingTxs<T>>::mutate(to_did, |pending_txs| pending_txs.extend(txs));
+                }
+            }
+            Ok(())
+        }
+
+        /// Handles an approved validator set change transaction proposal.
+        ///
+        /// NOTE: Extrinsics without `pub` are exported too. This function is declared as `pub` only
+        /// to test that it cannot be called from a wrong `origin`.
+        pub fn handle_change_validators(origin, new_validators: T::AccountId) -> DispatchResult {
+            let sender = ensure_signed(origin.clone())?;
+            ensure!(sender == Self::validators(), "should be called by the validator set account");
+            // Update the validator set.
+            <Validators<T>>::put(new_validators.clone());
+            // Remove the record of the ongoing proposal.
+            <ChangeValidatorsProposals<T>>::remove(&new_validators);
+            Self::deposit_event(RawEvent::ValidatorsChanged(new_validators));
+            Ok(())
+        }
+
+        /// Handles an approved bridge transaction proposal.
+        ///
+        /// NOTE: Extrinsics without `pub` are exported too. This function is declared as `pub` only
+        /// to test that it cannot be called from a wrong `origin`.
+        pub fn handle_bridge_tx(origin, bridge_tx: BridgeTx<T::AccountId>) -> DispatchResult {
+            let sender = ensure_signed(origin.clone())?;
+            ensure!(sender == Self::validators(), "should be called by the validator set account");
+            if let Some(PendingTx {
+                did,
+                bridge_tx,
+            }) = Self::issue(bridge_tx.clone())? {
+                <PendingTxs<T>>::mutate(did, |txs| txs.push(bridge_tx.clone()));
+                Self::deposit_event(RawEvent::Pending(PendingTx{
+                    did,
+                    bridge_tx: bridge_tx.clone()
+                }));
+            } else {
+                Self::deposit_event(RawEvent::Bridged(bridge_tx.clone()));
+            }
+            // Remove the record of the ongoing proposal.
+            <BridgeTxProposals<T>>::remove(&bridge_tx);
```

Comment:
Wouldn't this allow a single ethereum lock to be re minted on polymesh again and again?